### ISSUE TITLE
Visual Profiler displays incorrect CPU times

### ISSUE THREAD

**Issue by jitspoe** [godotengine/godot#97473]
### Tested versions

4.3

### System information

Windows 10, Vulkan forward +, Nvidia 3070

### Issue description

I've been doing a lot of profiling lately to try to find performance issues and added in Tracy support.  I've noticed the values profiled in the same area are VASTLY different.  I'm getting sub-100 fps, but the render time displayed by the built-in Visual Profiler is only showing around 2ms to render a viewport while Tracy shows ~8ms.

![image](https://github.com/user-attachments/assets/6c725198-b32f-45d9-88ee-507f01aa6677)

![image](https://github.com/user-attachments/assets/1d29d93b-b765-404a-b00b-f2dfe5f52fbd)

I shifted some of the RENDER_TIMESTAMP macros around as they didn't catch all the code in the function, but they still seem to be falling way short.

### Steps to reproduce

Start the visual profiler and note that the time for rendering does not match the time necessary to render the game at the current framerate.  Optionally get the tracy godot addon and add profile points where around things like RENDER_TIMESTAMP("> Render Viewports"); to compare values.

### Minimal reproduction project (MRP)

I don't think it's project specific, but you could use https://github.com/user-attachments/files/17140515/test_light_performance.zip

**Comment by clayjohn** [comment]
Are you using a build of 4.3 that includes https://github.com/godotengine/godot/pull/89398? I can see from the screenshot that you aren't on stable. So you might be missing things

**Comment by jitspoe** [comment]
Yep, I've got the RENDER_TIMESTAMP("Render OmniLight Shadows"); that was added there.  I'm not on stable because I'm doing my own builds.  The problem isn't the lack of profile points, it's that the values generated are incorrect.  I'll add a Tracy zone right next to an existing timestamp and get vastly different results, ex:

```
	for (int i = 0; i < sorted_active_viewports.size(); i++) {
		Viewport *vp = sorted_active_viewports[i];

		if (vp->last_pass != draw_viewports_pass) {
			continue; //should not draw
		}

		ZoneScopedN("Render Viewport");
		RENDER_TIMESTAMP("> Render Viewport " + itos(i));
```

As shown in the screenshot, it's like 2ms vs 8ms for the same block of code.  Tracy seems to be far more accurate.
